packages/statemanager/test/statelessVerkleStateManager.spec.ts

```diff
@@ -43,4 +52,115 @@ describe('StatelessVerkleStateManager: Kaustinen Verkle Block', () => {
       'should have correct codeHash'
     )
   })
+
+  it('put/delete/modify account', async () => {
+    const stateManager = new StatelessVerkleStateManager({ common })
+    stateManager.initVerkleExecutionWitness(block.executionWitness)
+
+    const address = new Address(randomBytes(20))
+
+    let check = await stateManager.getAccount(address)
+    assert.isUndefined(check, 'should return undefined for nonexistent account')
+
+    const account = Account.fromAccountData({
+      nonce: BigInt(2),
+    })
+    await stateManager.putAccount(address, account)
+
+    check = await stateManager.getAccount(address)
+    assert.deepEqual(check, account, 'should return correct account')
+
+    await stateManager.modifyAccountFields(address, {
+      nonce: BigInt(2),
+    })
+    check = await stateManager.getAccount(address)
+    assert.deepEqual(check, account, 'should return correct account')
+
+    await stateManager.deleteAccount(address)
+
+    check = await stateManager.getAccount(address)
+    assert.isUndefined(check, 'should return undefined for deleted account')
+  })
+
+  it('getTreeKeyFor* functions', async () => {
+    const stateManager = new StatelessVerkleStateManager({ common })
+    stateManager.initVerkleExecutionWitness(block.executionWitness)
+
+    const address = Address.fromString('0x9791ded6e5d3d5dafca71bb7bb2a14187d17e32e')
+    const stem = getStem(address, 0)
+
+    const balanceKey = stateManager.getTreeKeyForBalance(stem)
+    const nonceKey = stateManager.getTreeKeyForNonce(stem)
+    const codeHashKey = stateManager.getTreeKeyForCodeHash(stem)
+
+    const balanceRaw = stateManager['_state'][bytesToHex(balanceKey)]
+    const nonceRaw = stateManager['_state'][bytesToHex(nonceKey)]
+    const codeHash = stateManager['_state'][bytesToHex(codeHashKey)]
+
+    const account = await stateManager.getAccount(address)
+
+    assert.equal(
+      account.balance,
+      bytesToBigInt(hexToBytes(balanceRaw), true),
+      'should have correct balance'
+    )
+    assert.equal(
+      account.nonce,
+      bytesToBigInt(hexToBytes(nonceRaw), true),
+      'should have correct nonce'
+    )
+    assert.equal(bytesToHex(account.codeHash), codeHash, 'should have correct codeHash')
+  })
+
+  it(`copy()`, async () => {
+    const sm = new StatelessVerkleStateManager({
+      accountCacheOpts: {
+        type: CacheType.ORDERED_MAP,
+      },
+      storageCacheOpts: {
+        type: CacheType.ORDERED_MAP,
+      },
+      common,
+    })
+    sm.initVerkleExecutionWitness(block.executionWitness)
+
+    const smCopy = sm.shallowCopy()
```

Comment:
Our current shallowCopy implementations seems to have been overlooked, as it basically does nothing else than create a new StateManager with the same witness. 

I think we should implement `shallowCopy` similarly than what we have in `stateManager`. I'll tackle that this week